core/src/main/java/org/apache/brooklyn/core/config/BasicConfigKey.java

```diff
@@ -184,7 +192,7 @@ public String getDescription() {
     protected boolean reconfigurable;
     protected ConfigInheritance typeInheritance;
     protected ConfigInheritance runtimeInheritance;
-    protected Predicate<? super T> constraint;
+    protected List<Predicate<? super T>> constraints;
```

Comment:
This is persisted so should be extra careful with renames. When passing constraints from java the caller can wrap them in `and` if multiple, so this change is not needed. Suggest reverting.